src/mode_select.c

```
@@ -24,4 +32,127 @@ RunMode select_mode(void)
+
+	shared_data->buffer->is_ready = false; // this should be done in buffer instead of here
+}
+
+void *receive_thread(void *data)
+{
+	SharedData *shared_data = (SharedData *)data;
+	if (run_tcp_server(shared_data) != 0)
+	{
+		printf("\nError occured while running TCP server.\n");
+		return NULL;
+	}
```

Cppcheck (2.10-2)

[error] Found an exit path from function with non-void return type that has missing return statement

(missingReturn)